pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -705,8 +705,8 @@ func waitForPort(instanceLog *zap.SugaredLogger, netClient *gophercloud.ServiceC
 		return port.Status == "ACTIVE", nil
 	}
 
-	if err := wait.Poll(checkPeriod, checkTimeout, portIsReady); err != nil {
```

Comment:
Please don't introduce new background contexts, but instead add them to the callstacks where needed to hand them down to where they're needed.

Reply:
Updated